server/src/sas/CodeZoneManager.ts

```diff
@@ -251,57 +252,56 @@ export class CodeZoneManager {
   }
   private _token(line: number, col: number): TokenEx | null {
     let syntax = this._syntaxProvider.getSyntax(line);
-    const len = syntax.length;
-    let i = 1,
-      j = -1,
+    let foundSyntaxIdx = -1,
       type: Token["type"] = "text",
       currLine = line;
-    for (; i < len; i++) {
-      if (syntax[i].start > col) {
-        if (syntax[i - 1].start <= col) {
-          j = i - 1;
-          type = syntax[j].style;
+
+    for (let i = 0; i < syntax.length; i++) {
```

Comment:
@jingjiajie, would you please provide a detailed explanation for this change? It looks like a fundamental logic. I'd like to verify it carefully. Thank you.

Reply:
I remember this change is to fix the bug:

_There is no autocomplete if the first line is a comment. Try to type 'python' or 'print' next to 'proc'_

```
/*  */
proc 
```

Because if the starting one token of the whole document is a comment, __token()_ returned a null value. This change is to fix this bug.

The intention is only to fix that issue, and the remaining logics should keep the same behavior as before.